### PR TITLE
ARROW-16722: [CI][C++] Fix Minio failures specifying the Minio version to use

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by raulcd** [comment]
@pitrou using an older version of Minio fixes the errors (MinGW and AppVeyor builds are passing). There is a build error but does not seem related, maybe you can re-run it?
I am not sure about the fix using an old version but I don't know much about Minio and I haven't been able to find anything on their release notes that could point me to a possible fix either on our side or opening an issue: https://github.com/minio/minio/releases/tag/RELEASE.2022-06-02T02-11-04Z

**Comment by pitrou** [comment]
We should also do this on non-Windows platforms btw, there's a timeout on Travis-CI currently.

**Comment by raulcd** [comment]
> We should also do this on non-Windows platforms btw, there's a timeout on Travis-CI currently.

I have added a new commit using the same version for the rest of the platforms and also using a fixed version for the minio client. Waiting for CI.

**Review by jonkeane — APPROVED** [review]
I'm a fan of this, thanks for it!

File: ci/scripts/install_minio.sh [hunk 1]
```diff
@@ -51,15 +51,21 @@ fi
...
+# Use specific versions for minio server and client to avoid CI failures on new releases.
+minio_version="minio.RELEASE.2022-05-26T05-48-41Z"
+mc_version="mc.RELEASE.2022-05-09T04-08-26Z"
+
 if [[ ! -x ${prefix}/bin/minio ]]; then
-  url="https://dl.min.io/server/minio/release/${platform}-${arch}/minio"
+  url="https://dl.min.io/server/minio/release/${platform}-${arch}/archive/${minio_version}"
   echo "Fetching ${url}..."
   wget -nv -P ${prefix}/bin ${url}
+  mv ${prefix}/bin/${minio_version} ${prefix}/bin/minio
```
Comment (kou) [inline ci/scripts/install_minio.sh:62]:
```suggestion
  wget -nv --output-document ${prefix}/bin/minio ${url}
```

File: ci/scripts/install_minio.sh [hunk 2]
```diff
@@ -51,15 +51,21 @@ fi
...
   chmod +x ${prefix}/bin/minio
 fi
 if [[ ! -x ${prefix}/bin/mc ]]; then
-  url="https://dl.min.io/client/mc/release/${platform}-${arch}/mc"
+  url="https://dl.min.io/client/mc/release/${platform}-${arch}/archive/${mc_version}"
   echo "Fetching ${url}..."
   wget -nv -P ${prefix}/bin ${url}
+  mv ${prefix}/bin/${mc_version} ${prefix}/bin/mc
```
Comment (kou) [inline ci/scripts/install_minio.sh:69]:
```suggestion
  wget -nv --output-document ${prefix}/bin/mc ${url}
```

**Review by kou — APPROVED** [review]
+1

**Pull request merged**